<!--
    This .
-->
<launch>

    <!-- Defines device that will be used as camera -->
    <arg name="device" default="/dev/video0" />

    <!-- Defines FPS -->
    <arg name="fps" default="25" />

    <!-- Enables video streaming -->
    <arg name="stream" default="true" />

    <!-- Enables video recording -->
    <arg name="record" default="false" />

    <!-- launch video stream -->
    <include file="$(find video_stream_opencv)/launch/camera.launch">
        <!-- node name and ros graph name -->
        <arg name="camera_name" value="file" />
        <!-- means video device 0, /dev/video0 -->
        <arg name="video_stream_provider" value="/home/vladushked/Documents/hydro/dnn_gate_detector/run_detection_scripts/video/2.mp4" />
        <!-- set camera fps to (if the device allows) -->
        <arg name="set_camera_fps" value="30"/>
        <!-- set buffer queue size of frame capturing to -->
        <arg name="buffer_queue_size" value="100" />
        <!-- throttling the querying of frames to -->
        <arg name="fps" value="30" />
        <!-- setting frame_id -->
        <arg name="frame_id" value="image" />
        <!-- visualize on an image_view window the stream generated -->
        <arg name="visualize" value="false" />
    </include>

    <!-- Gate detector node -->
    <!-- output objects topic *node_name*/objects -->
    <!-- output image topic *node_name*/image -->
    <node name="object_detector" pkg="sauvc_vision" type="opencv_dnn_detector.py" output="screen" respawn="true">
        <param name="input_image_topic" value="/file/image_raw" />
        <param name="dnn_confidence_threshold" value="0.5" />
        <param name="enable_output_image_publishing" value="true" />
        <param name="package_name_with_net" value="sauvc_vision" />
    </node>

    <group if="$(arg record)">
        <!-- Video recorder node -->
        <node name="video_recorder" pkg="stingray_utils" type="video_recorder.py" output="screen" respawn="true">
            <param name="source_topic" value="/gate_detector/image" />
            <param name="output_width" value="640" />
            <param name="output_height" value="480" />
            <param name="output_fps" value="$(arg fps)" />
            <param name="output_format" value="h264" />
            <param name="record_dir" value="$(env HOME)/stingray_video_records/" />
        </node>
    </group>

    <group if="$(arg stream)">
        <!-- Video web server -->
        <!-- Stream will be available at http://<hostname>:8080/stream_viewer?topic=/device_camera/image_raw -->
        <node name="video_server" pkg="web_video_server" type="web_video_server" output="screen" respawn="true">
            <param name="server_threads" value="4" />
            <param name="ros_threads" value="10" />
        </node>
    </group>

</launch>